### PR TITLE
[Validator] add exception when intl component not found

### PR DESCRIPTION
**Pull request by ronfroy**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/Validator/Constraints/CountryValidator.php [hunk 1]
```diff
@@ -28,6 +29,10 @@ class CountryValidator extends ConstraintValidator
      */
     public function validate($value, Constraint $constraint)
     {
+        if (!class_exists('Intl')) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Validator/Constraints/CountryValidator.php]: should be Intl::class instead.

File: src/Symfony/Component/Validator/Constraints/CountryValidator.php [hunk 2]
```diff
@@ -28,6 +29,10 @@ class CountryValidator extends ConstraintValidator
...
+            throw new RuntimeException('symfony/intl is required to use the Country constraint');
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Validator/Constraints/CountryValidator.php]: missing dot at end of message, also I'd suggest to be a little bit more explicit (same in other messages below):
`The "symfony/intl" component is required to use the Country constraint.`

**Review by nicolas-grekas — COMMENTED** [review]
Good idea. Here are some comments.

**Review by nicolas-grekas — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

File: src/Symfony/Component/Validator/Constraints/CountryValidator.php [hunk 3]
```diff
@@ -40,6 +41,10 @@ public function validate($value, Constraint $constraint)
             throw new UnexpectedTypeException($value, 'string');
         }
 
+        if (!class_exists(Intl::class)) {
+            throw new RuntimeException('The "symfony/intl" component is required to use the Country constraint.');
```
Comment (chalasr) [inline src/Symfony/Component/Validator/Constraints/CountryValidator.php]: hmm, shouldn't it be a `LogicException` instead?

Comment (ronfroy) [reply src/Symfony/Component/Validator/Constraints/CountryValidator.php]: i think not, this is not a logic issue.

Comment (nicolas-grekas) [reply src/Symfony/Component/Validator/Constraints/CountryValidator.php]: looking at the code, we use LogicException quite often in similar situations, so could be worth for consistency at least (there are a few similar cases using RuntimeException, but they're not the majority and could be worth fixing)

Comment (javiereguiluz) [reply src/Symfony/Component/Validator/Constraints/CountryValidator.php]: I would have used `RuntimeException` too ... but if we use `LogicException` in similar situations, it's better to keep consistency. @ronfroy can you please update this? Thanks!

**Review by javiereguiluz — APPROVED** [review]

**Comment by ronfroy** [comment]
done

File: src/Symfony/Component/Validator/Constraints/CountryValidator.php (hunk 3 above)
Comment (ro0NL) [reply src/Symfony/Component/Validator/Constraints/CountryValidator.php]: how can we use `LogicException` elsewhere if this PR introduces it as a new class?

For this case we use the `RuntimeException` already :+1: 
https://github.com/symfony/symfony/blob/645b01659c2d54e264c181df8d28ddea5f8b619a/src/Symfony/Component/Validator/Constraints/ExpressionValidator.php#L56-L63

**Comment by nicolas-grekas** [comment]
can you update src/Symfony/Component/Validator/Constraints/BicValidator.php and src/Symfony/Component/Validator/Mapping/Factory/BlackHoleMetadataFactory.php to use the new exception also please?

**Comment by ronfroy** [comment]
@nicolas-grekas done

**Comment by nicolas-grekas** [comment]
and src/Symfony/Component/Validator/ValidatorBuilder.php

**Comment by ro0NL** [comment]
for consitency we should change Runtime to Logic in `EmailValidator, ExpressionValidator & ImageValidator` too :)

File: src/Symfony/Component/Validator/Tests/Mapping/Factory/BlackHoleMetadataFactoryTest.php [hunk 4]
```diff
@@ -17,7 +17,7 @@
 class BlackHoleMetadataFactoryTest extends TestCase
 {
     /**
-     * @expectedException \LogicException
+     * @expectedException \Symfony\Component\Validator\Exception\LogicException;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Validator/Tests/Mapping/Factory/BlackHoleMetadataFactoryTest.php]: not sure about the trailing;

File: src/Symfony/Component/Validator/ValidatorBuilder.php [hunk 5]
```diff
@@ -191,8 +192,8 @@ public function enableAnnotationMapping(Reader $annotationReader = null)
         }
 
         if (null === $annotationReader) {
-            if (!class_exists('Doctrine\Common\Annotations\AnnotationReader') || !class_exists('Doctrine\Common\Cache\ArrayCache')) {
-                throw new \RuntimeException('Enabling annotation based constraint mapping requires the packages doctrine/annotations and doctrine/cache to be installed.');
+            if (!class_exists(Doctrine\Common\Annotations\AnnotationReader::class) || !class_exists(Doctrine\Common\Cache\ArrayCache::class)) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Validator/ValidatorBuilder.php]: should be moved as "use" statements.

**Comment by nicolas-grekas** [comment]
> for consitency we should change Runtime to Logic in EmailValidator, ExpressionValidator & ImageValidator too :)

in another PR :)

**Comment by nicolas-grekas** [comment]
Thank you @ronfroy.

**Pull request merged**